sw/device/tests/dif/dif_hmac_sanitytest.c

```diff
@@ -0,0 +1,203 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_hmac.h"
+#include "sw/device/lib/arch/device.h"
+#include "sw/device/lib/base/log.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/common.h"
+#include "sw/device/lib/flash_ctrl.h"
+#include "sw/device/lib/runtime/check.h"
+#include "sw/device/lib/testing/test_main.h"
+#include "sw/device/lib/uart.h"
+
+#define HMAC0_BASE_ADDR 0x40120000
+#define MAX_FIFO_FILL 10
+
+static const char kData[142] =
+    "Every one suspects himself of at least one of "
+    "the cardinal virtues, and this is mine: I am "
+    "one of the few honest people that I have ever "
+    "known";
+
+static uint32_t kHmacKey[8] = {0x243f6a88, 0x85a308d3, 0x13198a2e, 0x03707344,
+                               0xa4093822, 0x299f31d0, 0x082efa98, 0xec4e6c89};
+
+static const uint32_t kExpectedShaDigest[8] = {
+    0xdc96c23d, 0xaf36e268, 0xcb68ff71, 0xe92f76e2,
+    0xb8a8379d, 0x426dc745, 0x19f5cff7, 0x4ec9c6d6};
+
+static const uint32_t kExpectedHmacDigest[8] = {
+    0xe4987b39, 0x3f83d390, 0xc2f3bbaf, 0x3195dbfa,
+    0x23fb480c, 0xb012ae5e, 0xf1394d28, 0x1940ceeb};
+
+/**
+ * Initialize the HMAC engine. Return `true` if the configuration is valid.
+ */
+static void test_setup(const dif_hmac_config_t *config, dif_hmac_t *hmac) {
+  dif_hmac_result_t res = dif_hmac_init(config, hmac);
+
+  CHECK(res != kDifHmacBadArg, "Invalid arguments encountered in HMAC init.");
+  CHECK(res == kDifHmacOk, "Unknown error encountered in HMAC init.");
```

Comment:
I don't think you should be duplicating these CHECK like this. You can clean this up in a followup.